### DISCUSSION TITLE
LayoutChanged gets fired on every mouseover event

### DISCUSSION THREAD

**Discussion by Rosentti** [AvaloniaUI/Avalonia#12872] [General]
Is this supposed to happen?
I was under the impression that LayoutChanged only fires when objects are **added/removed/shift around their order** in the layout tree. Is there a `Control` or `Layoutable` event I can use to only trigger on actual layout changes, or can I somehow filter out mouseovers, etc....

**Comment by timunie** [comment]
Probably SizeChanged event?

**Comment by Rosentti** [comment]
Hmm, I'm intending on doing retranslations/auto translating new objects, so SizeChanged would cause performance issues.

**Reply by timunie** [reply]
What exactly is the goal you have then? What is the input and what the desired output?

**Reply by Rosentti** [reply]
Retranslating when new objects are added to the visual/logical tree

**Reply by timunie** [reply]
what do you mean by retranslating? in terms of localization? In that case I'd use resx files or DynamicResources

**Reply by Rosentti** [reply]
I already have a pretty good translation system in place. I just want to add hooks to translate new objects that are added to the tree(s).

**Reply by timunie** [reply]
control.AttachedToVisualTree is what you may need to listen to. Maybe you can add it as a global eventhandler for all controls
